### ISSUE TITLE
Best way to structure large batch submission

### ISSUE THREAD

**Issue by avivajpeyi** [dfm/tess-atlas#132]
We probably dont need to worry about this atm (since we still have plenty of testing left), but itd be good to start thinking about this.

## Job estimates:

Total number of jobs: 4.5K

- `generation` step will generate a dir of ~5MB (~10 min)
- `analysis` step will generate a dir of ~20MB (~1-5Hrs)
- Rough total estimates:
    -    Total time ~ 10-20K CPU hours
    -   Total memory ~ 100 Gb
     -  Total # files ~ 60K (plots, logs, output)

## Potential setup strategy 
The easiest setup would be something like:

```
for each toi in list of tois:
    1. toi generation job: to download, setup dirs 
    2. toi analysis job: run analysis, save output
    3. toi cleanup job (?) : clear caches, delete junk, zip output
```

Given a large number of jobs, I may need to:
- stagger the job submission (only run in batches of 100 tois),
- create a clean-up post-processing job to zip up files once analysis jobs are done?

**Comment by avivajpeyi** [comment]
*tldr* we should be able to submit all jobs in one shot (no need to batch up jobs, or worry about memory)

From OzStar admins: 

> our typical starting disk quota for a project on /fred is 10TB and 1M files, so 0.1TB and 0.06M files isn't much at all.
> array jobs of 5000 or more are reasonably common on ozstar, so no worries there either. FWIW I think 10k is our array limit for 1 job.
> 
> anyway, I suggest that you just check your disk quota first with the 'quota' command, and then launch them all.
> should be finished in a day if you can get ~1000 jobs at once running... I suspect job throughput on the dm's will be the bottleneck.
> 
> regardless, I suggest you get an "array" of 1 to 10 jobs completely working and checked as much as you can before you submit an array of 4500. we see a surprising number of folks who just launch 1000 of anything and expect that they'll work flawlessly first time! 
> 
> as an optional tweak, I suppose you could submit a 3rd array of jobs that depends upon the 'analysis' jobs, and does a post-processing zip or tar as they finish? or that could be integrated into the end of the analysis script too. but as the numbers are small I don't think you really need to do this yet.

**Comment by avivajpeyi** [comment]
Going with the following method which was easy to setup with array jobs:
1. Run all toi generation step
2. Run all toi analysis

I had wanted to break this up such that we'd analyse as soon as the generation for one is completed, but I guess the above will have to do for now.

Closing for now
